tools/ansible-pr/run.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+set -x
+
+declare -a ignored_modules=(
+  gcp_backend_service
+  gcp_forwarding_rule
+  gcp_healthcheck
+  gcp_target_proxy
+  gcp_url_map
+)
+
+new_branch() {
+  branch_name=$1
+  git  checkout upstream/devel
+  git branch -D $branch_name
+  git chkecout -b $branch_name
+}
+
+get_all_modules() {
+  remote_name=$1
+  file_name=$remote_name
+  git fetch $remote_name
+  git checkout $remote_name/devel
+  git ls-files -- lib/ansible/modules/cloud/google/gcp_* | cut -d/ -f 6 | cut -d. -f 1 > $file_name
+  
+  for i in "${ignored_modules[@]}"; do
+    sed -i "/$i/d" $file_name
+  done
+}
+
+if ! which hub; then
+  echo "Please install the hub CLI"
+  exit 1
+fi
+
+if ! git remote -v | grep "origin"; then
+  echo "Please set the origin remote"
+  exit 1
+fi
+
+if ! git remote -v | grep "upstream.*git@github.com:ansible/ansible.git (fetch)" 2&>1 >/dev/null; then
+  git remote add upstream git@github.com:ansible/ansible.git
+fi
+
+git remote add magician git@github.com:modular-magician/ansible.git
+git fetch magician devel
+git fetch upstream devel
+echo "Remotes setup properly"
+
+# Create files with list of modules in a given branch.
+get_all_modules "upstream"
+get_all_modules "magician"
+
+# Split existing modules into sets of 23
+# Max 50 files per PR and a module can have 2 files (module + test)
+# 23 = 50/2 - 2 (to account for module_util files)
+split -l 23 upstream mm-bug
+
+for filename in mm-bug*; do
+  echo "Building a Bug Fix PR for $filename"
+  # Checkout all files that file specifies and create a commit.
+  git checkout upstream/devel
+  git branch -D bug_fixes$filename
+  git checkout -b bug_fixes$filename
+
+
+  while read p; do
+    git checkout magician/devel -- "lib/ansible/modules/cloud/google/$p.py"
+    git checkout magician/devel -- "test/integration/targets/$p"
+  done < $filename
+
+  git checkout magician/devel -- "lib/ansible/module_utils/gcp_utils.py"
+
+  git commit -m "Bug fixes for GCP modules" >/dev/null
+
+  # Create a PR message + save to file
+  ruby ../../tools/ansible-pr/generate_template.rb > bug_fixes$filename
+
+  # Create PR
+  git push origin HEAD --force
+  hub pull-request -b ansible/ansible:devel -F bug_fixes$filename
+  echo "Bug Fix PR built for $filename"
+done
+
+## Get list of new modules (in magician, not in upstream)
+comm -3 <(sort magician) <(sort upstream) > new_modules
+
+while read module; do
+  echo "Building a New Module PR for $module"
+  git checkout upstream/devel
+  git branch -D $module
+  git checkout -b $module
+
+  git checkout magician/devel -- "lib/ansible/modules/cloud/google/$module.py"
+  git checkout magician/devel -- "test/integration/targets/$module"
+
+  git checkout magician/devel -- "lib/ansible/module_utils/gcp_utils.py"
+
+  # Create a PR message + save to file
+  git commit -m "New Module: $module"
+  ruby ../../tools/ansible-pr/generate_template.rb --new-module-name $module > bug_fixes$filename
+
+  # Create PR
+  git push origin HEAD --force
```

Comment:
To which branch?

Reply:
In this case, it'll be `$module`

(Earlier, it would be `bug_fixes$filename`)

Would it be better practice to be explicit? Seems like it's just asking to break the script.

Reply:
From git push manpage:

```
           The <src> is often the name of the branch you would want to push,
           but it can be any arbitrary "SHA-1 expression", such as master~4 or
           HEAD (see gitrevisions(7)).
```

 seems like you're good!